libbeat/ebpf/watcher_linux.go

```diff
@@ -0,0 +1,167 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build linux
+
+package ebpf
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/elastic/ebpfevents"
+)
+
+var (
+	gWatcherOnce sync.Once
+	gWatcherErr  error
+	gWatcher     watcher
+)
+
+type client struct {
+	name    string
+	mask    EventMask
+	records chan ebpfevents.Record
+}
+
+type watcher struct {
+	sync.Mutex
+	ctx     context.Context
```

Comment:
Holding a context is generally seen as a bad thing to do. Is there a reason this needs to be stored in the struct?